File: source/cpp/242. Valid Anagram.cpp
#include <iostream>
#include <string_view>


class Solution {
public:
    bool isAnagram(std::string_view s, std::string_view t) {
        if (s.size() != t.size()) return false;

        int count[26] = { 0 };

        for (const auto& c : s) count[c - 'a']++;
        for (const auto& c : t) {
            if (count[c - 'a'] == 0) return false;
            count[c - 'a']--;
        }

        return true;
    }
};

File: source/python/242. Valid Anagram.py
from collections import Counter


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_counter = Counter(s)
        t_counter = Counter(t)

        return s_counter == t_counter


if __name__ == '__main__':
    s = Solution()

    assert s.isAnagram("anagram", "nagaram") == True
    assert s.isAnagram("rat", "car") == False

    print("All tests passed!")
